src/Compilers/Core/Portable/DiagnosticAnalyzer/ProgramFilesAnalyzerPathResolver.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis;
+
+/// <summary>
+/// This <see cref="IAnalyzerPathResolver"/> implementation is used to handle analyzers that
+/// exist in global install locations on Windows. These locations do not need to be shadow
+/// copied because they are read-only and are not expected to be updated. Putting this resolver
+/// before shadow copy will let them load in place.
+/// </summary>
+internal sealed class ProgramFilesAnalyzerPathResolver : IAnalyzerPathResolver
```

Comment:
ok.  this was stateless.  any concerns about having a singleton for this?  that would seem to match the pattern of other types in this pr.

Reply:
Originally this was stateful but eventually removed it. Will make it a singleton.